### PR TITLE
Feature/nfdiv 2175 existing or new case

### PR DESCRIPTION
**Pull request by benouaer**
### Change description ###
- add a new page (existing-application) where applicant selects existing or new invite case
- update case search logic to show page above if both an existing application and case invite exist
- add/update controllers to trigger SYSTEM_UNLINK_APPLICANT/SYSTEM_CANCEL_CASE_INVITE as required

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2175

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/test/functional/features/existing-application.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Given I click "Sign out"
+    Given I create a new user and login
+    And I've already completed the form using the fixture "jointApplicant1CompleteCase"
+    And I set applicant 2's email address
+    And I go to "/"
+    And I click "Send for review"
+    Then the page URL should be "/application-sent-for-review"
+    Given I click "Sign out"
+    And I login with applicant "2"
+    Then the page URL should be "/existing-application"
```
Comment (lara-mpt) [inline src/test/functional/features/existing-application.feature:20]: doesn't seem to like "I set applicant 2's email address" ?

File: src/main/steps/existing-application/content.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { CommonContent } from '../common/common.content';
+import { formattedCaseId } from '../common/content.utils';
+
+export enum existingOrNew {
+  Existing = 'existing',
+  New = 'new',
+}
+
+const en = ({ isDivorce, partner, isJointApplication, required, existingCaseId }) => {
+  const respondJoin = `${isJointApplication ? 'respond to' : 'join'}`;
```
Comment (danielbe11) [inline src/main/steps/existing-application/content.ts:13]: other way round?

File: src/main/steps/existing-application/content.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  return {
+    title: 'You have an existing application',
+    line1: `You already have an existing application for ${
+      isDivorce ? 'divorce' : 'to end your civil partnership'
+    }. Your existing application number is ${existingCaseId}`,
+    line2: `You are now being invited to ${respondJoin} a new application created by your ${partner}.
+            If you choose to ${respondJoin} it then you will lose access to your existing application.`,
+    newApplication: `I want to ${respondJoin} the new application`,
+    existingApplication: ' I want to continue with my existing application',
```
Comment (danielbe11) [inline src/main/steps/existing-application/content.ts:22]:
```suggestion
    existingApplication: 'I want to continue with my existing application',
```

File: src/main/steps/existing-application/content.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  New = 'new',
+}
+
+const en = ({ isDivorce, partner, isJointApplication, required, existingCaseId }) => {
+  const respondJoin = `${isJointApplication ? 'respond to' : 'join'}`;
+  return {
+    title: 'You have an existing application',
+    line1: `You already have an existing application for ${
+      isDivorce ? 'divorce' : 'to end your civil partnership'
+    }. Your existing application number is ${existingCaseId}`,
```
Comment (danielbe11) [inline src/main/steps/existing-application/content.ts:18]:
```suggestion
    }. Your existing application number is ${existingCaseId}.`,
```

File: src/main/steps/existing-application/content.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  },
+};
+
+const languages = {
+  en,
+  cy,
+};
+
+export const generateContent: TranslationFn = (content: CommonContent) => {
+  const existingCaseId = formattedCaseId(content.existingCaseId);
```
Comment (danielbe11) [inline src/main/steps/existing-application/content.ts:60]: Can we remove this and use the existingCaseId in the CommonContent? Then we'd need to use ${formattedCaseId(existingCaseId)} on L18.

File: src/main/steps/existing-application/content.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { isFieldFilledIn } from '../../app/form/validation';
+import { CommonContent } from '../common/common.content';
+import { formattedCaseId } from '../common/content.utils';
+
+export enum existingOrNew {
+  Existing = 'existing',
+  New = 'new',
+}
+
+const en = ({ isDivorce, partner, isJointApplication, required, existingCaseId }) => {
```
Comment (danielbe11) [inline src/main/steps/existing-application/content.ts:12]:
```suggestion
const en = ({ isDivorce, partner, isJointApplication, required, existingCaseId }: CommonContent) => {
```

**Review by danielbe11 — APPROVED** [review]

File: src/main/steps/index.test.ts [hunk 7]
```diff
@@ -52,6 +53,11 @@ describe('Steps', () => {
       const data = { gender: Gender.MALE };
       expect(getNextStepUrl(mockReq, data)).toBe(`${HAS_RELATIONSHIP_BROKEN_URL}?customQueryString`);
     });
+
+    it('returns / when the next step for applicant 2 is /applicant2/', () => {
+      mockReq.originalUrl = APPLICANT_2 + HUB_PAGE;
+      expect(getNextStepUrl(mockReq, {})).toBe(HOME_URL);
+    });
```
Comment (lara-mpt) [inline src/main/steps/index.test.ts:60]: could add this for respondent too?

**Review by lara-mpt — APPROVED** [review]

**Review by lara-mpt — APPROVED** [review]

**Pull request merged**